### DISCUSSION TITLE
How to test svelte components with mock urql client?

### DISCUSSION THREAD

**Discussion by Av1shay** [urql-graphql/urql#2343] [Q&A]
Hi,
I have a svelte component that using `operationStore()` as well as `getClient()` and `client.query()` from `@urql/svelte` and I follow these docs for testing the component: https://formidable.com/open-source/urql/docs/advanced/testing/
The docs are for react, there is easy since you just can pass providers and test the component. I'm not sure what's the approach for testing svelte component, I didn't find anything online.
I guess I need to use `setClient()` to set a mock client, but when I do something like this
```
describe('Test component', () => {
    const mockClient = {
        executeQuery: jest.fn(() => never),
        executeMutation: jest.fn(() => never),
        executeSubscription: jest.fn(() => never),
    };

    // @ts-ignore
    setClient(mockClient)

    test('test',()  => {
        render(Component)
        expect(mockClient.executeQuery).toBeCalledTimes(2);
    })
})
```
I get "Function called outside component initialization" for `setClient()` (since we can't use context API outside of a component in svelte).
What's the approach for this? If there is any.

Thanks in advance

**Comment by jonathanstanley** [comment]
I had a similar problem and I"m not sure there is a good solution.  This may be a limitation with the current implementation depending on context.  see https://github.com/FormidableLabs/urql/discussions/2113#discussioncomment-2411632

**Comment by Av1shay** [comment]
I ended setting a test context component like suggested here:
https://sveltesociety.dev/recipes/testing-and-debugging/unit-testing-svelte-component

Passing it the urql client from outside as prop, something like this
```
<script>
	import { setContext } from 'svelte';
        import { setClient } from '@urql/svelte';

	export let Component;
         export let urqlClient;
    
        setClient(urqlClient)
</script>

<svelte:component this={Component} />
```

```
test('test', ()  => {
    const urqlMockClient = {
        executeMutation: jest.fn(() => never),
    }
    const { getByRole } = render(ContextTest, {
        props: {
            Component: ComponentToBeTested,
            urqlClient: urqlMockClient,
        }
    ....
});
```
